{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>EC number</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>GH family</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CUX77898.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...</td>\n",
       "      <td>805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BAU88504.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...</td>\n",
       "      <td>805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAG06182.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAF60322.2</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAG06872.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MTATSDLIESLISYSWDDWQVTRQEARRVIAAIRNDNVPDATIAAL...</td>\n",
       "      <td>408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5891</th>\n",
       "      <td>AQR80106.1</td>\n",
       "      <td>3.2.1.92</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH171</td>\n",
       "      <td>MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5892</th>\n",
       "      <td>AQR84319.1</td>\n",
       "      <td>3.2.1.92</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH171</td>\n",
       "      <td>MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5893</th>\n",
       "      <td>QJF42043.1</td>\n",
       "      <td>3.2.1.92</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH171</td>\n",
       "      <td>MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5894</th>\n",
       "      <td>BAQ28034.1</td>\n",
       "      <td>4.2.1.179 3.2.1.- 3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH172</td>\n",
       "      <td>MSEITGLFKDLTKVKHARNGRLASWDQRGKNQDYWEIPAGESITLG...</td>\n",
       "      <td>452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5895</th>\n",
       "      <td>UMW87594.1</td>\n",
       "      <td>3.2.1.23</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH173</td>\n",
       "      <td>MKCNKLFVGAAALLLTLGSCSHSNGEWLTLIPGEPNPHAIECIGAE...</td>\n",
       "      <td>502</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5896 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              ID                  EC number  Taxonomy GH family  \\\n",
       "0     CUX77898.1                    3.2.1.-   Archaea       GH0   \n",
       "1     BAU88504.1                    3.2.1.-   Archaea       GH0   \n",
       "2     AAG06182.1                    3.2.1.-  Bacteria       GH0   \n",
       "3     AAF60322.2                    3.2.1.-  Bacteria       GH0   \n",
       "4     AAG06872.1                    3.2.1.-  Bacteria       GH0   \n",
       "...          ...                        ...       ...       ...   \n",
       "5891  AQR80106.1                   3.2.1.92  Bacteria     GH171   \n",
       "5892  AQR84319.1                   3.2.1.92  Bacteria     GH171   \n",
       "5893  QJF42043.1                   3.2.1.92  Bacteria     GH171   \n",
       "5894  BAQ28034.1  4.2.1.179 3.2.1.- 3.2.1.-  Bacteria     GH172   \n",
       "5895  UMW87594.1                   3.2.1.23  Bacteria     GH173   \n",
       "\n",
       "                                               Sequence  Length  \n",
       "0     MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...     805  \n",
       "1     MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...     805  \n",
       "2     MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...     438  \n",
       "3     MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...     438  \n",
       "4     MTATSDLIESLISYSWDDWQVTRQEARRVIAAIRNDNVPDATIAAL...     408  \n",
       "...                                                 ...     ...  \n",
       "5891  MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...     414  \n",
       "5892  MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...     414  \n",
       "5893  MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...     414  \n",
       "5894  MSEITGLFKDLTKVKHARNGRLASWDQRGKNQDYWEIPAGESITLG...     452  \n",
       "5895  MKCNKLFVGAAALLLTLGSCSHSNGEWLTLIPGEPNPHAIECIGAE...     502  \n",
       "\n",
       "[5896 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_vs_seq=pd.read_csv('non_GH_of_interest.csv') #read the csv file containing the length of each sequence as well, obtained from Get_EC_GH.ipynb\n",
    "#acc_vs_seq.rename( columns={'Unnamed: 0':'ID'}, inplace=True ) #rename the unnamed column as ID\n",
    "acc_vs_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>5896.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>522.023915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>203.940952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>64.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>368.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>514.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>669.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>999.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Length\n",
       "count  5896.000000\n",
       "mean    522.023915\n",
       "std     203.940952\n",
       "min      64.000000\n",
       "25%     368.000000\n",
       "50%     514.000000\n",
       "75%     669.000000\n",
       "max     999.000000"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_vs_seq.describe() #statistical description of the dataset\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>EC number</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>GH family</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CUX77898.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...</td>\n",
       "      <td>805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BAU88504.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...</td>\n",
       "      <td>805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAG06182.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAF60322.2</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAG06872.1</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH0</td>\n",
       "      <td>MTATSDLIESLISYSWDDWQVTRQEARRVIAAIRNDNVPDATIAAL...</td>\n",
       "      <td>408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5891</th>\n",
       "      <td>AQR80106.1</td>\n",
       "      <td>3.2.1.92</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH171</td>\n",
       "      <td>MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5892</th>\n",
       "      <td>AQR84319.1</td>\n",
       "      <td>3.2.1.92</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH171</td>\n",
       "      <td>MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5893</th>\n",
       "      <td>QJF42043.1</td>\n",
       "      <td>3.2.1.92</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH171</td>\n",
       "      <td>MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5894</th>\n",
       "      <td>BAQ28034.1</td>\n",
       "      <td>4.2.1.179 3.2.1.- 3.2.1.-</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH172</td>\n",
       "      <td>MSEITGLFKDLTKVKHARNGRLASWDQRGKNQDYWEIPAGESITLG...</td>\n",
       "      <td>452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5895</th>\n",
       "      <td>UMW87594.1</td>\n",
       "      <td>3.2.1.23</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>GH173</td>\n",
       "      <td>MKCNKLFVGAAALLLTLGSCSHSNGEWLTLIPGEPNPHAIECIGAE...</td>\n",
       "      <td>502</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5896 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              ID                  EC number  Taxonomy GH family  \\\n",
       "0     CUX77898.1                    3.2.1.-   Archaea       GH0   \n",
       "1     BAU88504.1                    3.2.1.-   Archaea       GH0   \n",
       "2     AAG06182.1                    3.2.1.-  Bacteria       GH0   \n",
       "3     AAF60322.2                    3.2.1.-  Bacteria       GH0   \n",
       "4     AAG06872.1                    3.2.1.-  Bacteria       GH0   \n",
       "...          ...                        ...       ...       ...   \n",
       "5891  AQR80106.1                   3.2.1.92  Bacteria     GH171   \n",
       "5892  AQR84319.1                   3.2.1.92  Bacteria     GH171   \n",
       "5893  QJF42043.1                   3.2.1.92  Bacteria     GH171   \n",
       "5894  BAQ28034.1  4.2.1.179 3.2.1.- 3.2.1.-  Bacteria     GH172   \n",
       "5895  UMW87594.1                   3.2.1.23  Bacteria     GH173   \n",
       "\n",
       "                                               Sequence  Length  \n",
       "0     MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...     805  \n",
       "1     MGLKRLLVLVVAVVVLSSGINFSWALGNRPSEGTGVAGKLSFKATF...     805  \n",
       "2     MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...     438  \n",
       "3     MNTYFDIPHRLVGKALYESYYDHFGQMDILSDGSLYLIYRRATEHV...     438  \n",
       "4     MTATSDLIESLISYSWDDWQVTRQEARRVIAAIRNDNVPDATIAAL...     408  \n",
       "...                                                 ...     ...  \n",
       "5891  MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...     414  \n",
       "5892  MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...     414  \n",
       "5893  MRKTIFAFLTGLMMFGTITAASASPDSKNQTAKKPKVQTGIDTLLP...     414  \n",
       "5894  MSEITGLFKDLTKVKHARNGRLASWDQRGKNQDYWEIPAGESITLG...     452  \n",
       "5895  MKCNKLFVGAAALLLTLGSCSHSNGEWLTLIPGEPNPHAIECIGAE...     502  \n",
       "\n",
       "[5896 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_vs_seq=acc_vs_seq[acc_vs_seq.Length <1000] #removing sequences more than 1000 and less than 60 amino acids long\n",
    "acc_vs_seq=acc_vs_seq[acc_vs_seq.Length >60]\n",
    "acc_vs_seq.reset_index(inplace = True, drop = True) #resetting the index, as after dropping there are missing index values\n",
    "acc_vs_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "multifasta file with no length outliers is successfully saved\n"
     ]
    }
   ],
   "source": [
    "#saving the sequences to multifasta after length trimmed\n",
    "xx=(acc_vs_seq.ID.count())\n",
    "with open('non_GH_of_interest.fasta','w',encoding='utf-8') as otp:\n",
    "    for i in range(xx):\n",
    "        acc_id=acc_vs_seq.iloc[i,0]+'&'+acc_vs_seq.iloc[i,1]+'&'+acc_vs_seq.iloc[i,3]\n",
    "        seq=acc_vs_seq.iloc[i,4]\n",
    "        otp.write('>'+acc_id+'\\n'+seq+'\\n')\n",
    "print('multifasta file with no length outliers is successfully saved')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
